src/SSHDebugPS/Docker/DockerConnection.cs

```diff
@@ -12,15 +13,92 @@ namespace Microsoft.SSHDebugPS.Docker
 {
     internal class DockerConnection : PipeConnection
     {
+        #region Statics
+
+        internal const string SshPrefixRegex = @"^[Ss]{2}[Hh]\s*=\s*";
+        internal const string SshPrefix = "ssh=";
+        internal const string DockerHostPrefixRegex = @"^host\s*=\s*";
+        internal const string DockerHostPrefix = "host=";
+        internal const char Separator = ';';
+
+        internal static string CreateConnectionString(string containerName, string remoteConnectionName, string hostName)
+        {
+            string connectionString = containerName;
+            if (!string.IsNullOrWhiteSpace(remoteConnectionName))
+            {
+                connectionString += Separator + SshPrefix + remoteConnectionName;
+            }
+
+            if (!string.IsNullOrWhiteSpace(hostName))
+            {
+                connectionString += Separator + DockerHostPrefix + hostName;
+            }
+
+            return connectionString;
+        }
+
+        internal static bool TryConvertConnectionStringToSettings(string connectionString, out DockerContainerTransportSettings settings, out Connection remoteConnection)
+        {
+            remoteConnection = null;
+            settings = null;
+
+            string containerName = string.Empty;
+            string hostName = string.Empty;
+            bool invalidString = false;
+
+            // Assume format is <containername>;ssh=<sshconnection>;host=<dockerhostvalue> or some mixture
+            string[] connectionStrings = connectionString.Split(Separator);
+
+            if (connectionStrings.Length <= 3 && connectionStrings.Length > 0)
+            {
+                Regex SshRegex = new Regex(SshPrefixRegex);
+                Regex dockerHostRegex = new Regex(DockerHostPrefixRegex);
+
+                foreach (var item in connectionStrings)
+                {
+                    string segment = item.Trim(' ');
+                    if (SshRegex.IsMatch(segment))
+                    {
+                        Match match = SshRegex.Match(segment);
+                        remoteConnection = ConnectionManager.GetSSHConnection(segment.Substring(match.Length));
+                    }
+                    else if (dockerHostRegex.IsMatch(segment))
+                    {
+                        Match match = dockerHostRegex.Match(segment);
+                        hostName = segment.Substring(match.Length);
+                    }
+                    else if (segment.Contains("="))
+                    {
+                        invalidString = true;
+                    }
+                    else
+                    {
+                        Debug.Assert(string.IsNullOrWhiteSpace(containerName), "containerName should be empty");
```

Comment:
>Debug.Assert(string.IsNullOrWhiteSpace(containerName), "containerName should be empty"); [](start = 24, length = 88)

Set `invalidString = true;` instead of just asserting?